### PR TITLE
`azurerm_spring_cloud_service` - support for the `marketplace` property

### PR DESCRIPTION
**Pull request by ms-henglu**
Hi reviewers,

I understand it's not recommended  to use `Optional+Computed`, but the service will add a default value for the existing services, without `Computed`, it would be a breaking change to the existing users.

```
=== RUN   TestAccSpringCloudService_basic
=== PAUSE TestAccSpringCloudService_basic
=== CONT  TestAccSpringCloudService_basic
--- PASS: TestAccSpringCloudService_basic (415.59s)
=== RUN   TestAccSpringCloudService_update
=== PAUSE TestAccSpringCloudService_update
=== CONT  TestAccSpringCloudService_update
--- PASS: TestAccSpringCloudService_update (1001.05s)
=== RUN   TestAccSpringCloudService_complete
=== PAUSE TestAccSpringCloudService_complete
=== CONT  TestAccSpringCloudService_complete
--- PASS: TestAccSpringCloudService_complete (553.41s)
=== RUN   TestAccSpringCloudService_virtualNetwork
=== PAUSE TestAccSpringCloudService_virtualNetwork
=== CONT  TestAccSpringCloudService_virtualNetwork
=== CONT  TestAccSpringCloudService_virtualNetwork
--- PASS: TestAccSpringCloudService_virtualNetwork (413.76s)
=== RUN   TestAccSpringCloudService_requiresImport
=== PAUSE TestAccSpringCloudService_requiresImport
=== CONT  TestAccSpringCloudService_requiresImport
--- PASS: TestAccSpringCloudService_requiresImport (334.43s)
=== RUN   TestAccSpringCloudService_serviceRegistry
=== PAUSE TestAccSpringCloudService_serviceRegistry
=== CONT  TestAccSpringCloudService_serviceRegistry
--- PASS: TestAccSpringCloudService_serviceRegistry (650.43s)
=== RUN   TestAccSpringCloudService_buildAgentPool
=== PAUSE TestAccSpringCloudService_buildAgentPool
=== CONT  TestAccSpringCloudService_buildAgentPool
--- PASS: TestAccSpringCloudService_buildAgentPool (596.56s)
=== RUN   TestAccSpringCloudService_zoneRedundant
=== PAUSE TestAccSpringCloudService_zoneRedundant
=== CONT  TestAccSpringCloudService_zoneRedundant
--- PASS: TestAccSpringCloudService_zoneRedundant (413.91s)
=== RUN   TestAccSpringCloudService_containerRegistry
=== PAUSE TestAccSpringCloudService_containerRegistry
=== CONT  TestAccSpringCloudService_containerRegistry
--- PASS: TestAccSpringCloudService_containerRegistry (585.85s)
=== RUN   TestAccSpringCloudService_marketplace
=== PAUSE TestAccSpringCloudService_marketplace
=== CONT  TestAccSpringCloudService_marketplace
--- PASS: TestAccSpringCloudService_marketplace (451.26s)
```

### CODE REVIEW

File: website/docs/r/spring_cloud_service.html.markdown [hunk 1]
```diff
@@ -72,6 +72,8 @@ The following arguments are supported:
 
 * `sku_name` - (Optional) Specifies the SKU Name for this Spring Cloud Service. Possible values are `B0`, `S0` and `E0`. Defaults to `S0`. Changing this forces a new resource to be created.
 
+* `marketplace` - (Optional) A `marketplace` block as defined below. This field is applicable only for Spring Cloud Service with enterprise tier.
```
Comment (manicminer) [inline website/docs/r/spring_cloud_service.html.markdown:75]:
```suggestion
* `marketplace` - (Optional) A `marketplace` block as defined below. Can only be specified when `sku` is set to `E0`.
```

File: website/docs/r/spring_cloud_service.html.markdown [hunk 2]
```diff
@@ -136,6 +138,16 @@ The `default_build_service` block supports the following:
 
 ---
 
+The `marketplace` block supports the following:
+
+* `plan` - (Required) Specifies the plan id of the 3rd Party Artifact that is being procured.
```
Comment (manicminer) [inline website/docs/r/spring_cloud_service.html.markdown:143]:
```suggestion
* `plan` - (Required) Specifies the plan ID of the 3rd Party Artifact that is being procured.
```

File: website/docs/r/spring_cloud_service.html.markdown [hunk 3]
```diff
@@ -136,6 +138,16 @@ The `default_build_service` block supports the following:
...
+
+* `publisher` - (Required) Specifies the publisher id of the 3rd Party Artifact that is being bought.
```
Comment (manicminer) [inline website/docs/r/spring_cloud_service.html.markdown:145]:
```suggestion
* `publisher` - (Required) Specifies the publisher ID of the 3rd Party Artifact that is being procured.
```

**Review by manicminer — APPROVED** [review]
Thanks @ms-henglu, pending tests this LGTM with some adjustments to the docs wording,

**Comment by ms-henglu** [comment]
Hi @manicminer ,

Thanks for the code review, the tests are passing now, would you please merge it, thanks!
image

**Pull request merged**